src/main/java/seedu/address/model/person/Person.java

```diff
@@ -35,9 +39,27 @@ public Person(Name name, Phone phone, Email email, Address address,
         this.phone = phone;
         this.email = email;
         this.address = address;
-        // protect internal tags from changes in the arg list
+        // protect internal groups from changes in the arg list
+        this.groupTags = new UniqueGroupList(groupTags);
+        // protect internal preferences from changes in the arg list
+        this.prefTags = new UniquePreferenceList(prefTags);
+        // protect internal orders from changes in the arg list
+        this.orders = new UniqueOrderList();
+    }
+
+    public Person(Name name, Phone phone, Email email, Address address, Set<Group> groupTags,
```

Comment:
Just to confirm, is this constructor called when converting XMLAdaptedPerson from storage to Model Person Object?

Reply:
This overloaded constructor was actually added because we might not want to add orders to a Person on creation of a new contact. So I left the original constructor intact. The overloaded one will be used when the Person needs to be edited and already has orders assigned to him (like groups/pref need to be synced, orders will be too).

Reply:
Alright, I understand. This PR is ready to be merged!

Reply:
Thanks!